core/metacling/src/TClingCallFunc.cxx

```diff
@@ -268,7 +281,17 @@ void TClingCallFunc::make_narg_ctor(const unsigned N, ostringstream &typedefbuf,
          callbuf << "*(" << type_name.c_str() << "**)args["
                  << i << "]";
       } else {
+         // By-value construction: Figure out if the type can be copy-constructed. This is tricky and cannot be done in
+         // a fully reliable way, also because std::vector<T> always defines a copy constructor, even if the type T is
+         // only moveable. As a heuristic, we only check if the copy constructor is deleted, or would be if implicit.
+         bool Move = IsCopyConstructorDeleted(QT);
```

Comment:
I presume we don't want to move by default. Was there any discussion why?

Reply:
It's passing by value, so calling a function should by default not empty out the original argument. However, if it's the only way to do it, we should pretend this is what the user wanted.